src/runtime/internal/loop.js

```diff
@@ -1,28 +1,24 @@
-import { raf } from './environment';
+import { raf } from './environment.js';
 
-export interface Task {
-	abort(): void;
-	promise: Promise<void>;
-}
-
-type TaskCallback = (now: number) => boolean | void;
-type TaskEntry = { c: TaskCallback; f: () => void };
-
-const tasks = new Set<TaskEntry>();
+const tasks = new Set();
```

Comment:
Maybe we should still have internal types? Below, i see task.f() being called, which prolly won't give intellisense.

In future, when have new contributors, small things liek these help a lot to make the structure clearer.

Saying this from exeprience cuz i recently revamped the REPL and it didn't have any internal types, which made understanding the flow extremely hard

Reply:
Ha, don't worry, we're not deleting any types. This PR is in very early draft phase and we're using an automated tool to do the gruntwork. It's a bit imprecise, so we have to munge things together afterwards.